apis/policies/v1alpha2/clusteradmissionpolicy_types.go

```diff
@@ -190,7 +206,7 @@ type ClusterAdmissionPolicyStatus struct {
 //+kubebuilder:resource:scope=Cluster
 //+kubebuilder:storageversion
 //+kubebuilder:printcolumn:name="Mutating",type=boolean,JSONPath=`.spec.mutating`,description="Whether the policy is mutating"
-//+kubebuilder:printcolumn:name="Active",type=boolean,JSONPath=`.status.policyActive`,description="Whether the policy is active and receiving admission reviews"
+//+kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.policyStatus`,description="Status of the policy"
 type ClusterAdmissionPolicy struct {
```

Comment:
(Opening comment thread here as I can't open on the specific code, as it is unchanged)

From @raulcabello:
> @viccuad I can't create policies without settings, I get The ClusterAdmissionPolicy "privileged-pods" is invalid: spec.settings: Invalid value: "null": spec.settings in body must be of type object: "null"
But if I remove the mutating webhook then I can create them

I can reproduce it, and I don't know from where it comes from. The policy spec is unchaged, and settings is marked as `optional` and `omitempty`.

Reply:
I don't see why the webhook would reject it, here I see it as approved:
```
2021-09-17T15:37:26.369Z    DEBUG    controller-runtime.webhook.webhooks    received request    {"webhook": "/mutate-policies-kubewarden-io-v1alpha2-clusteradmissionpolicy", "UID": "9d210836-8494-4835-999b-c3d89fa48604", "kind": "policies.kubewarden.io/v1alpha2, Kind=ClusterAdmission
Policy", "resource": {"group":"policies.kubewarden.io","version":"v1alpha2","resource":"clusteradmissionpolicies"}}                                                                                                                                                                         
2021-09-17T15:37:26.369Z    INFO    clusteradmissionpolicy-resource    default    {"name": "psp-capabilities-no-settings"}                                                                                                                                                                  
2021-09-17T15:37:26.369Z    DEBUG    controller-runtime.webhook.webhooks    wrote response    {"webhook": "/mutate-policies-kubewarden-io-v1alpha2-clusteradmissionpolicy", "code": 200, "reason": "", "UID": "9d210836-8494-4835-999b-c3d89fa48604", "allowed": true}
```

Reply:
marking .spec.settings as `//+nullable` makes it work. but I don't see why is it needed, given that is marked as `//+optional` already.

Reply:
yes with `//+nullable` works! It make sense to add nullable here. You can have an optional field with a default value, so it would never be null

Reply:
it seems that in 1.20 (where I am), objects with empty values get pruned automatically if they don't have `nullable: true` (see https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable).

Committed it in https://github.com/kubewarden/kubewarden-controller/pull/76/commits/5c01067db40feadbb6f1dca16b4430ad4457ba2d.

Reply:
Thanks! @viccuad great work!